### PR TITLE
dotCMS/core#15976 NG: Update html render un angular to prevent use of innerhtml

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/models/dot-contentlets-events.model.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { DotPageContent } from '@portlets/dot-edit-page/shared/models/dot-page-content.model';
+import { DotPageContainer } from '@portlets/dot-edit-page/shared/models/dot-page-container.model';
+
+export interface DotContentletEvent {
+    name: string;
+}
+
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/models/dot-contentlets-events.model.ts]: I think maybe we can do by something like:

export interface DotContentletEvent< T > {
    name: string;
    data: T
}

export interface DotContentletEventRelocate extends DotContentletEvent<DotRelocatePayload> {
}

export interface DotContentletEventSave extends DotContentletEvent< DotContentletPayload > {

}

File: src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts [hunk 2]
```diff
@@ -501,30 +521,31 @@ export class DotEditContentHtmlService {
...
         // Add innerHTML to a plain so we can get the HTML nodes later
         const div = doc.createElement('div');
         div.innerHTML = html;
 
-        const contentDivWrapper = div.children[0];
+        return <HTMLElement>div.children[0];
+    }
+
+    private appendNewContentlets(contentletEl: HTMLElement, html: string): void {
+        console.log('appendNewContentlets');
```
Comment (hmoreras) [inline src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts]: remove !

File: src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts [hunk 3]
```diff
@@ -719,21 +748,46 @@ export class DotEditContentHtmlService {
         this.addVtlEditMenu(contentletEl);
     }
 
-    private renderRelocatedContentlet(relocateInfo: any): void {
+    private renderRelocatedContentlet(relocateInfo: DotRelocatePayload): void {
+        console.log('renderRelocatedContentlet');
```
Comment (hmoreras) [inline src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts]: remove

File: src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts [hunk 4]
```diff
@@ -719,21 +748,46 @@ export class DotEditContentHtmlService {
...
+
+        relocateInfo.container = relocateInfo.container || {
+            identifier: container.dataset.dotIdentifier,
+            uuid: container.dataset.dotUuid
+        };
 
         this.dotContainerContentletService
             .getContentletToContainer(relocateInfo.container, relocateInfo.contentlet)
             .subscribe((contentletHtml: string) => {
+                console.log('contentletHtml', contentletHtml);
```
Comment (hmoreras) [inline src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts]: remove

**Pull request merged**